From 19295a7acbddddecd3e0d46d6ee0f22c3af852e0 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Wed, 11 Sep 2019 14:01:19 +0300
Subject: [PATCH 1/6] libc: Add global __LIBC definition while building libc.

 It will be used to limit visibility for uses outside libc context.
 No functional change yet.

In-discussion-with: swilder
---
 lib/libc/Makefile              | 1 +
 lib/libc/include/port_before.h | 2 ++
 lib/libc_rtld/Makefile         | 2 +-
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/libc/Makefile b/lib/libc/Makefile
index 00b7d2091e..2e995b9f8e 100644
--- a/lib/libc/Makefile
+++ b/lib/libc/Makefile
@@ -6,6 +6,7 @@ SHLIB_MAJOR= 8
 SHLIBDIR?= /lib
 CFLAGS+= -I${.CURDIR}/include -I${.OBJDIR} -I${.CURDIR}/${MACHINE_ARCH}
 CFLAGS+= -DNLS
+CFLAGS+= -D__LIBC
 CLEANFILES+=tags
 PRECIOUSLIB=	yes
 
diff --git a/lib/libc/include/port_before.h b/lib/libc/include/port_before.h
index 9bd12899d3..f243ae202e 100644
--- a/lib/libc/include/port_before.h
+++ b/lib/libc/include/port_before.h
@@ -3,7 +3,9 @@
 #ifndef _PORT_BEFORE_H_
 #define _PORT_BEFORE_H_
 
+#ifdef __LIBC
 #define _LIBC		1
+#endif
 #define DO_PTHREADS	1
 #define USE_KQUEUE	1
 
diff --git a/lib/libc_rtld/Makefile b/lib/libc_rtld/Makefile
index f1a9b2d855..32fb36dd51 100644
--- a/lib/libc_rtld/Makefile
+++ b/lib/libc_rtld/Makefile
@@ -10,7 +10,7 @@ CFLAGS+=-I${.CURDIR}/../../include
 CFLAGS+=-I${.CURDIR}/../libc/${MACHINE_ARCH}
 CFLAGS+=-I${.OBJDIR}
 CFLAGS+=-D__thread=
-CFLAGS+=-D__LIBC_RTLD
+CFLAGS+=-D__LIBC -D__LIBC_RTLD
 
 PRECIOUSLIB=	yes
 
-- 
2.23.0

